app/javascript/packages/document-capture/components/in-person-full-address-search.tsx

```diff
@@ -141,17 +141,32 @@ function FullAddressSearch({
     validatedZipCodeFieldRef,
   } = useUspsLocations(locationsURL);
 
+  const inputChangeHandlerForZipCode =
+    <T extends HTMLElement & { value: string }>(input) =>
+    (event: React.ChangeEvent<T>) => {
+      const { target } = event;
+      const scrubbedZip = (target.value).replace(/[^0-9]/g, '');
+      if (scrubbedZip.length > 5) {
+        let zipcodePieces : string[] = [];
+        zipcodePieces.push(scrubbedZip.substring(0,5));
+        zipcodePieces.push(scrubbedZip.substring(5));
+        input(zipcodePieces.join('-'));
+      } else {
+        input(scrubbedZip);
+      }
+    };
+
   const inputChangeHandler =
     <T extends HTMLElement & { value: string }>(input) =>
     (event: React.ChangeEvent<T>) => {
       const { target } = event;
-      input(target.value);
+      input((target.value).trimStart());
     };
```

Comment:
I used trimStart() rather than trim(). An input of "123 Somewhere Drive" would turn into "123SomewhereDrive" with trim because when "123 " was typed in, the space would be removed. By only using trim() in handleSearch, a user can enter an input of "   " and would not hit a validation error.